src/Symfony/Component/HttpClient/Internal/HttplugWaitLoop.php

```diff
@@ -142,4 +147,17 @@ public function createPsr7Response(ResponseInterface $response, bool $buffer = f
 
         return $psrResponse->withBody($body);
     }
+
+    private function getReasonPhrase(ResponseInterface $response): ?string
```

Comment:
because the class is internal, I propose to make it public static and reuse call it from Psr18Client

I also propose to return the empty string when no reason phrase is found, and simplify the logic on the caller sides: no need for the complexity about arguments anymore

Reply:
The problem is that the psr17factory reacts different depending on the number of parameters

Reply:
There it checks for the amount of parameters:

https://github.com/Nyholm/psr7/blob/3cb4d163b58589e47b35103e8e5e6a6a475b47be/src/Factory/Psr17Factory.php#L25-L28

Reply:
oops, thanks for the link